rest/server_context.go

```diff
@@ -652,52 +649,29 @@ func (sc *ServerContext) _getOrAddDatabaseFromConfig(ctx context.Context, config
 		useViews = true
 	}
 
-	type indexInitData struct {
-		scopeAndCollection base.ScopeAndCollectionName
-		indexSet           db.CollectionIndexesType
-		datastore          base.DataStore
-	}
-
-	collectionsRequiringIndexes := make([]indexInitData, 0)
-
-	// initDataStore is a function to initialize Views or GSI indexes for a datastore
-	initDataStore := func(ds base.DataStore, metadataIndexes db.CollectionIndexesType, name base.ScopeAndCollectionName) (err error) {
-		if useViews {
-			viewErr := db.InitializeViews(ctx, ds)
-			if viewErr != nil {
-				return viewErr
-			}
-			return nil
-		}
-
-		indexInit := indexInitData{
-			scopeAndCollection: name,
-			indexSet:           metadataIndexes,
-			datastore:          ds,
-		}
-		collectionsRequiringIndexes = append(collectionsRequiringIndexes, indexInit)
-		return nil
-	}
-
+	hasDefaultCollection := false
 	collectionsRequiringResync := make([]base.ScopeAndCollectionName, 0)
 	if len(config.Scopes) > 0 {
 		if !bucket.IsSupported(sgbucket.BucketStoreFeatureCollections) {
 			return nil, errCollectionsUnsupported
 		}
 
-		hasDefaultCollection := false
 		for scopeName, scopeConfig := range config.Scopes {
```

Comment:
Take or leave this, but it has been confusing to me that we loop over the datastores here for collection require resync. It doesn't quite feel like this belongs in the same function as `buildCollectionIndexData` because this code is actually attempting to read teh datastore and writing to it with `InitSyncInfo`. It could include an extra parameter that represents `hasDefaultCollectionConfigured`, though that would be technically redundant information.

I might change the code to work something more like this as separate operations:

1. do a check for `if len(config.Scopes) > 0 && if !bucket.IsSupported(sgbucket.BucketStoreFeatureCollections)` to return an error.
2. call `buildCollectionIndexData` to get the dsNames and loop over them to call `GetAndWaitUntilDataStoreReady` and call set `InitSyncInfo`. `InitSyncInfo` is skipped for the case were `_default._default` is only stored for metadata (see above for `hasDefaultCollectionConfigured` variable suggestion).

I think we want to leave `InitSyncInfo` before the `DatabaseInitManager`, because capella lets you start a resync before the indexes have been created. The resync manager will block on to wait for the principal collections to be enabled.

I find this code _really_ confusing especially since it seems to be duplicating the work in the database init manager, which was sort of what you tried to resolve in the first place.

Reply:
I agree that the refactor isn't going far enough to make `DatabaseInitManger` cover _everything_ a database needs (views/`SyncInfo`), and potentially move into `db` package so we can reduce the responsibility of `_getOrAddDatabaseFromConfig` and the `rest` package to keep `db` a bit more standalone, but it will blow up scope that isn't nessesary for implementing the collection-index enhancements I'm working towards.